lib/jar.ts

```diff
@@ -138,11 +141,13 @@ async function getMavenPackageInfo(sha1: string): Promise<MavenPackageInfo> {
             ),
           );
         }
-
         if (res.response.docs.length > 1) {
-          debug('Got multiple results for sha1, only returning first one');
+          const sha1Target = path.parse(targetPath).base;
+          debug('Got multiple results for sha1, looking for', sha1Target);
+          res.response.docs.map((doc) =>
+            sha1Target.includes(doc.g) ? resolve(doc) : 0,
```

Comment:
Does result come back sorted? If no, maybe there is some kind of `score` to find the best matching one?

Reply:
I am checking if other elements as tags may help